tests/Credentials/CredentialProviderTest.php

```diff
@@ -369,6 +370,68 @@ public function testCreatesFromProcessCredentialWithFilename()
         $this->assertEquals('bar', $creds->getSecretKey());
     }
 
+    public function testCreatesFromProcessCredentialWithFilenameParameterOverSharedFilename()
+    {
+        $dir = $this->clearEnv();
+        $ini = <<<EOT
+[baz]
+credential_process = echo '{"AccessKeyId":"foo","SecretAccessKey":"bar", "Version":1}'
+EOT;
+        file_put_contents($dir . '/mycreds', $ini);
+        putenv('HOME=' . dirname($dir));
+        putenv("AWS_SHARED_CREDENTIALS_FILE={$dir}/badfilename");
+
+
+        $creds = call_user_func(
+            CredentialProvider::process('baz', $dir . '/mycreds')
+        )->wait();
+        unlink($dir . '/mycreds');
+        $this->assertEquals('foo', $creds->getAccessKeyId());
+        $this->assertEquals('bar', $creds->getSecretKey());
+    }
+
+    public function testCreatesFromProcessCredentialWithSharedFilename()
```

Comment:
You've covered calling the providers directly, but should there also be a test that the default provider ends up using the new ENV variable as well?